rero_ils/modules/items/serializers/csv.py

```diff
@@ -0,0 +1,127 @@
+
+"""Item serializers."""
+
+import csv
+
+from invenio_i18n.ext import current_i18n
+from invenio_records_rest.serializers.csv import CSVSerializer, Line
+
+from rero_ils.filter import format_date_filter
+from rero_ils.modules.documents.api import search_document_by_pid
+from rero_ils.modules.documents.utils import title_format_text_head
+from rero_ils.modules.loans.api import search_loans_for_item
+from rero_ils.modules.locations.api import LocationsSearch
+
+
+class ItemCSVSerializer(CSVSerializer):
+    """Serialize and filter item circulation status."""
+
+    def transform_search_hit(
+        self, pid, record_hit, links_factory=None, **kwargs
+    ):
+        """Transform search result hit into an intermediate representation."""
+        hit = self.preprocess_search_hit(
+            pid, record_hit, links_factory=links_factory, **kwargs
+        )
+        return hit
+
+    def preprocess_search_hit(self, pid, record_hit, links_factory=None,
+                              **kwargs):
+        """Prepare a record hit from Elasticsearch for serialization."""
+        record = record_hit['_source']
+        item_pid = pid.pid_value
+
+        # process location
+        locations_map = kwargs.get('locations_map')
+        record['location_name'] = locations_map[
+            record.get('location').get('pid')]
+
+        # retrieve and process document
+        document = search_document_by_pid(record['document']['pid'])
+        record['document_title'] = title_format_text_head(document.title,
+                                                          with_subtitle=True)
+        record['document_type'] = document.type
+
+        # get loans information
+        loans_count, loans = search_loans_for_item(item_pid)
+        record['loans_count'] = loans_count
+        if loans_count > 0:
+            # get first loan
+            loan = loans[0]['_source']
+            record['last_transaction_date'] = format_date_filter(
+                loan['transaction_date'],
+                format='short_date',
+                locale=current_i18n.locale.language,
+            )
+
+        record['created'] = format_date_filter(
+            record['_created'],
+            format='short_date',
+            locale=current_i18n.locale.language,
+        )
+
+        # prevent csv key error
+        # TODO: find other way to prevent csv key error
+        del(record['status'])
+        del (record['type'])
+
+        return dict(
+            record,
+        )
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
+                         item_links_factory=None):
+        """Serialize a search result.
+
+        :param pid_fetcher: Persistent identifier fetcher.
+        :param search_result: Elasticsearch search result.
+        :param links: Dictionary of links to add to response.
+        :param item_links_factory: Factory function for record links.
+        """
+        records = []
+        locations_map = {}
```

Comment:
what is the purpose of this?

Reply:
I loads all locations to avoid to call location api to retrieve the  location for each record. 
1000 records to process => 1000 api calls